api/server.go

```diff
@@ -98,14 +99,25 @@ func (s *Server) walletAddressHandler(w http.ResponseWriter, req *http.Request,
 
 func (s *Server) walletAddressesHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
 	addresses := s.w.Addresses()
-	offset, limit := offsetLimit(req.URL.Query(), len(addresses))
-	writeJSON(w, WalletAddresses(addresses[offset:limit]))
+	begin, end := beginEnd(req.URL.Query(), len(addresses))
+	writeJSON(w, WalletAddresses(addresses[begin:end]))
```

Comment:
I'd probably inline `beginEnd`, so that you can have better error messages:

```go
addresses := s.w.Addresses()
start, end := 0, len(addresses)
if v := req.FormValue("start"); v != "" {
    i, err := strconv.Atoi(v)
    if err != nil || i < 0 || i > len(addresses) {
        http.Error(w, "invalid start value", http.StatusBadRequest)
        return
    }
    start = i
}
if v := req.FormValue("end"); v != "" {
    i, err := strconv.Atoi(v)
    if err != nil || i < 0 || i < start {
        http.Error(w, "invalid end value", http.StatusBadRequest)
        return
    } else if i > len(addresses)
        i = len(addresses)
    }
    end = i
}
writeJSON(w, WalletAddresses(addresses[begin:end]))
```